### ISSUE TITLE
Request to OpenStack Glance has broken percent-encoding on URL argument

### ISSUE THREAD

**Issue by majewsky** [fog/fog#3572]
The following code,

```ruby
my_image = fog_image.images.find_by_name("Foo Bar")
# where fog_image is an instance of Fog::Image::OpenStack::Real
```

using Fog v1.29.0, generates the following HTTP request:

```
GET http://glance:9292/v1.1/images/detail?name=Foo%2520Bar
```

As you can see, the space in the image name is malencoded as `%2520` rather than `%20`, thus the lookup fails and `my_image` is nil.

The remote site is a Glance in a Devstack (Kilo release).

**Comment by TerryHowe** [comment]
Maybe something like this: https://github.com/fog/fog/pull/3573

**Comment by geemus** [comment]
closed by #3573
